core/src/main/java/feign/QueryMapEncoder.java

```diff
@@ -34,55 +31,10 @@
    */
   Map<String, Object> encode(Object object);
 
-  class Default implements QueryMapEncoder {
-
-    private final Map<Class<?>, ObjectParamMetadata> classToMetadata =
-        new HashMap<Class<?>, ObjectParamMetadata>();
-
-    @Override
-    public Map<String, Object> encode(Object object) throws EncodeException {
-      try {
-        ObjectParamMetadata metadata = getMetadata(object.getClass());
-        Map<String, Object> fieldNameToValue = new HashMap<String, Object>();
-        for (Field field : metadata.objectFields) {
-          Object value = field.get(object);
-          if (value != null && value != object) {
-            fieldNameToValue.put(field.getName(), value);
-          }
-        }
-        return fieldNameToValue;
-      } catch (IllegalAccessException e) {
-        throw new EncodeException("Failure encoding object into query map", e);
-      }
-    }
-
-    private ObjectParamMetadata getMetadata(Class<?> objectType) {
-      ObjectParamMetadata metadata = classToMetadata.get(objectType);
-      if (metadata == null) {
-        metadata = ObjectParamMetadata.parseObjectType(objectType);
-        classToMetadata.put(objectType, metadata);
-      }
-      return metadata;
-    }
-
-    private static class ObjectParamMetadata {
-
-      private final List<Field> objectFields;
-
-      private ObjectParamMetadata(List<Field> objectFields) {
-        this.objectFields = Collections.unmodifiableList(objectFields);
-      }
-
-      private static ObjectParamMetadata parseObjectType(Class<?> type) {
-        List<Field> fields = new ArrayList<Field>();
-        for (Field field : type.getDeclaredFields()) {
-          if (!field.isAccessible()) {
-            field.setAccessible(true);
-          }
-          fields.add(field);
-        }
-        return new ObjectParamMetadata(fields);
-      }
-    }
+  /**
+   * @deprecated use {@link BeanQueryMapEncoder} instead.
```

Comment:
If they use `BeanQueryMapEncoder` instead, their stuff will break. They should use `FieldQueryMapEncoder` as the new replacement for this default encoder.

Reply:
You might want to add some `@see` comments for the currently known implementations to the `QueryMapEncoder` interface javadoc though.

Reply:
Agreed. I recommend removing the deprecation notice and use the javadoc references.

Reply:
Well, since `Default` visibility was default, I don't think anyone is using it.

So, deprecation notice can be safely removed.